src/v/cluster/rm_stm.cc

```diff
@@ -1888,6 +2077,31 @@ rm_stm::apply_snapshot(stm_snapshot_header hdr, iobuf&& tx_ss_buf) {
     iobuf_parser data_parser(std::move(tx_ss_buf));
     if (hdr.version == tx_snapshot::version) {
         data = reflection::adl<tx_snapshot>{}.from(data_parser);
+    } else if (hdr.version == tx_snapshot_v2::version) {
+        auto data_v2 = reflection::adl<tx_snapshot_v2>{}.from(data_parser);
```

Comment:
serde?

Reply:
time constrains, we'll do it after release https://github.com/redpanda-data/redpanda/issues/6877